gtp2ogs.js

```diff
@@ -1017,10 +1021,36 @@ class Game {
         if (this.state.phase != 'play')
             return;
 
-	let sendMove = (moves) => {  this.uploadMove(moves[0]);  };
-	let sendPass = ()      => {  this.uploadMove({'x': -1});  };
+	let sendPass = () => {  this.uploadMove({'x': -1});  };
+	let doing_handicap = (this.state.free_handicap_placement && this.state.handicap > 1 &&
+			      this.state.moves.length < this.state.handicap);
 	
-	this.getBotMoves("genmove " + this.my_color, sendMove, sendPass);
+	if (!doing_handicap) {  // Regular genmove ...
+	    let sendMove = (moves) => {  this.uploadMove(moves[0]);  };
+	    this.getBotMoves("genmove " + this.my_color, sendMove, sendPass);
+	    return;
+	}
+	
+	// Already have handicap stones ? Return next one.
+	if (this.handicap_moves.length) {
+	    this.uploadMove(this.handicap_moves.shift());
```

Comment:
What is the advantage of storing and sending them one at a time over sending them all straight away? The latter also seemed to work and would allow us to skip the move storage.

Reply:
Oh ok, let's send them all together then. I guess i was trying to keep it the same from server side point of view.